ironfish-cli/src/utils/transaction.ts

```diff
@@ -14,6 +14,80 @@ import {
   UnsignedTransaction,
 } from '@ironfish/sdk'
 import { CliUx } from '@oclif/core'
+import { ProgressBar } from '../types'
+
+export class TransactionTimer {
+  private logger: Logger
+  private progressBar: ProgressBar | undefined
+  private startTime: number | undefined
+  private estimateInMs: number
+  private timer: NodeJS.Timer | undefined
+
+  constructor(
+    spendPostTime: number,
+    raw: RawTransaction,
+    logger?: Logger,
+    options: {
+      displayEstimate?: boolean
+    } = {
+      displayEstimate: true,
+    },
+  ) {
+    this.logger = logger ?? createRootLogger()
+    this.estimateInMs = Math.max(Math.ceil(spendPostTime * raw.spends.length), 1000)
+    options.displayEstimate && this.displayEstimate()
+  }
+
+  private displayEstimate() {
+    this.logger.log(
+      `Time to send: ${TimeUtils.renderSpan(this.estimateInMs, {
+        hideMilliseconds: true,
+      })}`,
+    )
+  }
+
+  startTimer() {
+    this.progressBar = CliUx.ux.progress({
+      format: '{title}: [{bar}] {percentage}% | {estimate}',
+    }) as ProgressBar
+
+    this.startTime = Date.now()
+
+    this.progressBar.start(100, 0, {
+      title: 'Sending the transaction',
+      estimate: TimeUtils.renderSpan(this.estimateInMs, { hideMilliseconds: true }),
+    })
+
+    this.timer = setInterval(() => {
+      if (!this.progressBar || !this.startTime) {
+        return
+      }
+      const durationInMs = Date.now() - this.startTime
+      const timeRemaining = this.estimateInMs - durationInMs
+      const progress = Math.round((durationInMs / this.estimateInMs) * 100)
+
+      this.progressBar.update(progress, {
+        estimate: TimeUtils.renderSpan(timeRemaining, { hideMilliseconds: true }),
```

Comment:
If `timeRemaining` is negative, this will show a negative estimate. Is there a way to show "unknown" or maybe "soon" instead?

Reply:
The behavior currently involves "counting up" after the countdown is complete. I talked through some options with Jason when initially implementing this and this seemed like the best one. This PR is only a refactor, so I'll leave as is but I'll continue to think through more options.